From 8ec4fcb3339e1d81390e456b7b052744e4af7c4d Mon Sep 17 00:00:00 2001
From: Brendan Le Foll <brendan.le.foll@intel.com>
Date: Fri, 2 Sep 2016 12:00:28 +0100
Subject: [PATCH] intel_galileo_rev_g.c: use pincmd to set OUT_HIGH instead of
 value

The i/o expander looses connection as soon as direction is set so use direction
OUT_HIGH instead of setting value after direction. Closes #573

Signed-off-by: Brendan Le Foll <brendan.le.foll@intel.com>
---
 src/x86/intel_galileo_rev_g.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/x86/intel_galileo_rev_g.c b/src/x86/intel_galileo_rev_g.c
index d0fe8d0..35533c0 100644
--- a/src/x86/intel_galileo_rev_g.c
+++ b/src/x86/intel_galileo_rev_g.c
@@ -808,9 +808,9 @@ mraa_intel_galileo_gen2()
     b->pins[18].aio.mux[0].pincmd = PINCMD_SET_DIRECTION;
     b->pins[18].aio.mux[0].pin = 57;
     b->pins[18].aio.mux[0].value = MRAA_GPIO_IN;
-    b->pins[18].aio.mux[1].pincmd = PINCMD_SET_OUT_VALUE;
+    b->pins[18].aio.mux[1].pincmd = PINCMD_SET_DIRECTION;
     b->pins[18].aio.mux[1].pin = 60;
-    b->pins[18].aio.mux[1].value = 1;
+    b->pins[18].aio.mux[1].value = MRAA_GPIO_OUT_HIGH;
     b->pins[18].aio.mux[2].pincmd = PINCMD_SET_VALUE;
     b->pins[18].aio.mux[2].pin = 78;
     b->pins[18].aio.mux[2].value = 0;
@@ -819,9 +819,9 @@ mraa_intel_galileo_gen2()
     b->pins[18].gpio.mux[0].pincmd = PINCMD_SET_DIRECTION;
     b->pins[18].gpio.mux[0].pin = 57;
     b->pins[18].gpio.mux[0].value = MRAA_GPIO_IN;
-    b->pins[18].gpio.mux[1].pincmd = PINCMD_SET_OUT_VALUE;
+    b->pins[18].gpio.mux[1].pincmd = PINCMD_SET_DIRECTION;
     b->pins[18].gpio.mux[1].pin = 60;
-    b->pins[18].gpio.mux[1].value = 1;
+    b->pins[18].gpio.mux[1].value = MRAA_GPIO_OUT_HIGH;
     b->pins[18].gpio.mux[2].pincmd = PINCMD_SET_VALUE;
     b->pins[18].gpio.mux[2].pin = 78;
     b->pins[18].gpio.mux[2].value = 1;
@@ -845,9 +845,9 @@ mraa_intel_galileo_gen2()
     b->pins[19].aio.mux[0].pincmd = PINCMD_SET_DIRECTION;
     b->pins[19].aio.mux[0].pin = 59;
     b->pins[19].aio.mux[0].value = MRAA_GPIO_IN;
-    b->pins[19].aio.mux[1].pincmd = PINCMD_SET_OUT_VALUE;
+    b->pins[19].aio.mux[1].pincmd = PINCMD_SET_DIRECTION;
     b->pins[19].aio.mux[1].pin = 60;
-    b->pins[19].aio.mux[1].value = 1;
+    b->pins[19].aio.mux[1].value = MRAA_GPIO_OUT_HIGH;
     b->pins[19].aio.mux[2].pincmd = PINCMD_SET_VALUE;
     b->pins[19].aio.mux[2].pin = 79;
     b->pins[19].aio.mux[2].value = 0;
@@ -856,9 +856,9 @@ mraa_intel_galileo_gen2()
     b->pins[19].gpio.mux[0].pincmd = PINCMD_SET_DIRECTION;
     b->pins[19].gpio.mux[0].pin = 59;
     b->pins[19].gpio.mux[0].value = MRAA_GPIO_IN;
-    b->pins[19].gpio.mux[1].pincmd = PINCMD_SET_OUT_VALUE;
+    b->pins[19].gpio.mux[1].pincmd = PINCMD_SET_DIRECTION;
     b->pins[19].gpio.mux[1].pin = 60;
-    b->pins[19].gpio.mux[1].value = 1;
+    b->pins[19].gpio.mux[1].value = MRAA_GPIO_OUT_HIGH;
     b->pins[19].gpio.mux[2].pincmd = PINCMD_SET_VALUE;
     b->pins[19].gpio.mux[2].pin = 79;
     b->pins[19].gpio.mux[2].value = 1;
-- 
2.1.4

